SecurityPkg/Library/AuthVariableLib/AuthService.c

```diff
@@ -2000,15 +2206,14 @@ VerifyTimeBasedPayload (
   //
   // Example generated with: https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Manual_process
   //
+  Status = FindHashAlgorithmIndex (SigData, SigDataSize, &HashAlgId);
   if ((Attributes & EFI_VARIABLE_TIME_BASED_AUTHENTICATED_WRITE_ACCESS) != 0) {
-    if (  (  (SigDataSize >= (13 + sizeof (mSha256OidValue)))
-          && (  ((*(SigData + 1) & TWO_BYTE_ENCODE) != TWO_BYTE_ENCODE)
-             || (CompareMem (SigData + 13, &mSha256OidValue, sizeof (mSha256OidValue)) != 0)))
-       && (  (SigDataSize >= (32 + sizeof (mSha256OidValue)))
-          && (  ((*(SigData + 20) & TWO_BYTE_ENCODE) != TWO_BYTE_ENCODE)
-             || (CompareMem (SigData + 32, &mSha256OidValue, sizeof (mSha256OidValue)) != 0))))
-    {
-      return EFI_SECURITY_VIOLATION;
+    //
+    // A SigDataSize of zero, is a special case. This means that the variable is being deleted.
+    // In this case, the payload is empty and the signature will not be verified.
+    //
+    if (EFI_ERROR (Status) && (SigDataSize != 0)) {

```

Comment:
I think we still need to add MU_CHANGE tag for your own change? This as is seems to have combined the upstream change and yours?